build/erpnext-worker/Dockerfile

```diff
@@ -0,0 +1,44 @@
+FROM bitnami/python:latest-prod
+
+RUN useradd -ms /bin/bash frappe
+WORKDIR /home/frappe/frappe-bench
+RUN install_packages \
+    git \
+    wkhtmltopdf \
+    mariadb-client \
+    gettext-base \
+    wget \
+    # for PDF
+    libssl-dev \
+    fonts-cantarell \
+    xfonts-75dpi \
+    xfonts-base \
+    # For psycopg2
+    libpq-dev \
+    build-essential
+
+# Install wkhtmltox correctly
+RUN wget https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/0.12.5/wkhtmltox_0.12.5-1.stretch_amd64.deb
+RUN dpkg -i wkhtmltox_0.12.5-1.stretch_amd64.deb && rm wkhtmltox_0.12.5-1.stretch_amd64.deb
+
+RUN mkdir -p apps logs commands
+
+RUN virtualenv env \
+    && . env/bin/activate \
+    && cd apps \
+    && git clone --depth 1 -o upstream https://github.com/frappe/frappe \
+    && git clone --depth 1 -o upstream https://github.com/frappe/erpnext \
+    && pip3 install --no-cache-dir -e /home/frappe/frappe-bench/apps/frappe \
+    && pip3 install --no-cache-dir -e /home/frappe/frappe-bench/apps/erpnext
+
+COPY build/common/commands/* /home/frappe/frappe-bench/commands/
+COPY build/common/common_site_config.json.template /opt/frappe/common_site_config.json.template
```

Comment:
I guess these commands should be ran by `frappe` user. Otherwise permissions are not right. For example I encountered an error for `frappe-schedule` container:

```
frappe-schedule_1        | Traceback (most recent call last):
frappe-schedule_1        |   File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/scheduler.py", line 45, in enqueue_events_for_all_sites
frappe-schedule_1        |     enqueue_events_for_site(site=site)
frappe-schedule_1        |   File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/scheduler.py", line 67, in enqueue_events_for_site
frappe-schedule_1        |     frappe.logger(__name__).debug('Access denied for site {0}'.format(site))
frappe-schedule_1        |   File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1547, in logger
frappe-schedule_1        |     return get_logger(module or 'default', with_more_info=with_more_info)
frappe-schedule_1        |   File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/logger.py", line 18, in get_logger
frappe-schedule_1        |     LOG_FILENAME, maxBytes=100000, backupCount=20)
frappe-schedule_1        |   File "/opt/bitnami/python/lib/python3.7/logging/handlers.py", line 148, in __init__
frappe-schedule_1        |     BaseRotatingHandler.__init__(self, filename, mode, encoding, delay)
frappe-schedule_1        |   File "/opt/bitnami/python/lib/python3.7/logging/handlers.py", line 55, in __init__
frappe-schedule_1        |     logging.FileHandler.__init__(self, filename, mode, encoding, delay)
frappe-schedule_1        |   File "/opt/bitnami/python/lib/python3.7/logging/__init__.py", line 1087, in __init__
frappe-schedule_1        |     StreamHandler.__init__(self, self._open())
frappe-schedule_1        |   File "/opt/bitnami/python/lib/python3.7/logging/__init__.py", line 1116, in _open
frappe-schedule_1        |     return open(self.baseFilename, self.mode, encoding=self.encoding)
frappe-schedule_1        | PermissionError: [Errno 13] Permission denied: '/home/frappe/frappe-bench/logs/frappe.log'
```

Reply:
This is not related to virtualenv being created as root.

`sites` volume directory is created as root on host machine. non root uid from the container is not able to access it.

Also while using in docker I'd like logs to be on stdout and I'll get them from `docker logs`. May be it needs a PR and config in frappe.

Anything you can think of?

Reply:
This issue isn't actually related to `sites`, but to `logs`, which is created on image build.

I didn't check what it wanted to write in the logs directory. Next time I encounter this I'll post it.
I guess that as a workaround (to suppress the error) `logs` could be created as `frappe` and left as is until we figure out how to route all messages to docker. Maybe there is some command line switch for that?

Reply:
There we go:
```
[DEBUG] 2020-03-02 00:42:07,913 | /home/frappe/frappe-bench/apps/frappe/frappe/utils/scheduler.py:
Site: *****
Queued events for site *****
```

Where ***** is the site name.

Reply:
for now I set ownership of `logs` directory during container entry